src/decorators/log.decorator.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+export function log(level?: number) {
```

Comment:
@bajtos I can set a log level by binding it and pass in a log level for a function into the decorator - but how would I check the log level bound by the app here to configure this accordingly? Same issue as PR#1
---
Just thinking about this some more, Is this where a provide would come into this? ... Wouldn't it be easier to have direct access to the value from Context here?

Reply:
Your `@log` decorator should set some sort of a metadata - either using https://github.com/strongloop/loopback-next/blob/210d8a13a60991c391433b9f9ea1885cca135ee0/packages/context/src/reflect.ts#L183 or by modifying the decorated class/method directly.

Then you need to write a custom sequence action that the application needs to call from its sequence, this sequence action will retrieve the metadata set by `@log` decorator and decide whether to print the log.

I think you are making this unnecessary complicated.

Reply:
I agree with @bajtos. The proposed decorator implementation is too invasive as it patches the original controller method and returns a new property descriptor.